bolt/lib/Target/AArch64/AArch64MCPlusBuilder.cpp

```diff
@@ -277,6 +277,48 @@ class AArch64MCPlusBuilder : public MCPlusBuilder {
     }
   }
 
+  MCPhysReg
+  getRegUsedAsCallDest(const MCInst &Inst,
+                       bool &IsAuthenticatedInternally) const override {
```

Comment:
I'm wondering if this could be adapt so that it only needs to handle indirect calls?
That would make the switch statement simpler, and also easier to maintain, because it won't need to handle all branch instructions.

For example, at the moment, it seems the switch statement is not handling the newly introduced (in armv9.5) Compare and Branch instructions, see [https://developer.arm.com/documentation/ddi0602/2024-09/Base-Instructions/CB-cc---register---Compare-registers-and-branch-](https://developer.arm.com/documentation/ddi0602/2024-12/Base-Instructions/CB-cc---register---Compare-registers-and-branch-)

My understanding is that only indirect calls need to be checked, not direct calls.

Reply:
Thanks for pointing this out! Non-indirect control flow instructions are inherently irrelevant here.

Reply:
Opened a separate PR #133227 on updating `isIndirectCall()`.